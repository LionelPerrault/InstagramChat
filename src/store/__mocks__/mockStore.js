import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
