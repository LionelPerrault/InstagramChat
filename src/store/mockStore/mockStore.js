import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'
import Post from './mock-reducers/Post'
import Hashtag from './mock-reducers/Hashtag'
import Setting from './mock-reducers/Setting'
import Message from './mock-reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message
}

const middlewares = [
  promise(), thunk
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
