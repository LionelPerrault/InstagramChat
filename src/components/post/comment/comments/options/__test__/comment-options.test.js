import React from 'react'
import MockDataElement from '../../../../../../utils/__mocks__/mock-dataElement'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../../../../store/__mocks__/mockStore'
import CommentOptions, { PureCommentTools } from '../comment-options'
import { shallow } from 'enzyme'

describe('CommentOptions Component', () => {
  let dataElement
  const mockFn = jest.fn()

  beforeEach(() => (dataElement = MockDataElement()))

  afterEach(() => dataElement.remove())

  const commentDetails = {
    comment_id: 4,
    comment_by: 7,
    type: 'text',
    text: 'some comment',
    commentSrc: 'instagram_comment_1518972851259.jpg',
  }

  const comp = (extraDetails = {}) => (
    <Provider store={mockStore}>
      <CommentOptions
        commentDetails={{
          ...commentDetails,
          ...extraDetails,
        }}
        decrementComments={mockFn}
        updateCommentText={mockFn}
      />
    </Provider>
  )

  it('should match snapshot with null', () => {
    const tree = create(comp()).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot and show toggle icon', () => {
    // when comment is mine
    const tree = create(
      comp({
        comment_by: 24,
      })
    ).toJSON()
    expect(tree).toMatchSnapshot()

    // when iam the admin
    dataElement.setAttribute('data-isadmin', 'true')
    const tree2 = create(comp()).toJSON()
    expect(tree2).toMatchSnapshot()
  })

  it('should show options', () => {
    const wrapper = shallow(
      <PureCommentTools
        commentDetails={{
          ...commentDetails,
          comment_by: 24,
        }}
        decrementComments={mockFn}
        updateCommentText={mockFn}
      />
    )

    wrapper.find('.toggle_options').simulate('click')
    expect(wrapper.find('.comments_options').exists()).toBe(true)
  })

  it('should not show options', () => {
    const wrapper = shallow(
      <PureCommentTools
        commentDetails={commentDetails}
        decrementComments={mockFn}
        updateCommentText={mockFn}
      />
    )
    expect(wrapper.find('.comments_options').exists()).toBe(false)
  })
})
