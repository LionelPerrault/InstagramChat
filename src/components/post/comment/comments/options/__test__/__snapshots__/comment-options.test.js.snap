// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentOptions Component should match snapshot and show toggle icon 1`] = `
Array [
  <span
    className="toggle_options"
    data-tip="Options"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      expand_more
    </i>
  </span>,
  <div
    className="__react_component_tooltip place-top type-dark "
    data-id="tooltip"
  >
    
  </div>,
]
`;

exports[`CommentOptions Component should match snapshot and show toggle icon 2`] = `
Array [
  <span
    className="toggle_options"
    data-tip="Options"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      expand_more
    </i>
  </span>,
  <div
    className="__react_component_tooltip place-top type-dark "
    data-id="tooltip"
  >
    
  </div>,
]
`;

exports[`CommentOptions Component should match snapshot with null 1`] = `
<div
  className="__react_component_tooltip place-top type-dark "
  data-id="tooltip"
>
  
</div>
`;
